joomla/joomla-cms
# [3.9] Empty lines in log, non functional defer option

### Summary of Changes
formatLine() function should return formatted text, not write it to file.

### Testing Instructions
Open Joomla 3.9 beta 3 site, and try few times invalid login. Then go to your logs dir, and see the error.log.

### Expected result
Every attempt should be on single line.

### Actual result
Every attempt is followed by an empty line.

Apply the patch and try few invalid logins again, see the log again.

### Side effects
As side effect this PR also makes new defer option really functional. As formatLine() function did not return the value, but writes it directly to file, the file was opened multiple times, not once as desired.
